1-JavaScript/Week3/homworkSolition/ex2-mondaysWorth.test.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const mondayTasks = [
+  {
+    name: 'Daily standup',
+    duration: 30,
+  },
+  {
+    name: 'Feature discussion',
+    duration: 120,
+  },
+  {
+    name: 'Development time',
+    duration: 240,
+  },
+  {
+    name: 'Talk to different members from the product team',
+    duration: 60,
+  },
+];
+
+const hourlyRate = 25;
+function computeEarnings(mondayTasks,hourlyRate) {
+  const newmondayTasks = [...mondayTasks]
+  const totEarn = newmondayTasks.map((task) => { return task.duration/60 * 25 }).
+  reduce((acc,val) => {return acc + val},0).toFixed(2)
+  const totEarnS = `€${totEarn}`
+  return totEarnS
+
+}
```

Comment:
Nice nice nice, I like how you're using the built-in array methods! 💪 Also great to see that you are chaining `.map` and `.reduce` together!

Some tips:
1. You don't necessarily need to spread the existing array into a new one, because `.map` creates a new array and does not modify the original one 🙂 
2. Instead of hardcoding 25, you can use the `hourlyRate` variable instead. This is passed as an argument so you can use it within the function.
3. You can just return the string with the total earnings instead of making a new variable and returning that 😄 
4. There is actually a way to do all of this in one line with `.reduce` 😲 If you like the extra challenge, you can try it but it's not necessary!

```suggestion
function computeEarnings(mondayTasks,hourlyRate) {
  const totEarn = newmondayTasks.map((task) => task.duration / 60 * hourlyRate).
  reduce((acc,val) => acc + val,0).toFixed(2)

   return `€${totEarn}`
}
```